signac/project.py

```diff
@@ -1398,7 +1429,7 @@ def _remove_persistent_cache_file(self):
             if error.errno != errno.ENOENT:
                 raise error
 
-    def update_cache(self):
+    def update_cache(self, validate=True):
```

Comment:
If we can, I would like to keep users from having to think about "should I validate?" in any public APIs. When would we expect this to be called with `validate=False`? I see this is used in a single test below -- is that test needed? I haven't looked closely at what it's covering.

Reply:
I added this only for the purposes of that test which assumes that `update_cache` will populate the cache will statepoint dict that does not match the key. I also don't like adding this option - I think the cache should always contain valid id: statepoint mappings.

I could remove this validate argument and also remove the test that it breaks.

Reply:
This is the test that fails when `update_cache` validates entries:
https://github.com/glotzerlab/signac/blob/1de715530749018c334f8244a1d4cd25b61ef5a6/tests/test_project.py#L512-L516
The `job.init()` on line 516 fails because `update_cache` reads the corrupted statepoint and raises `JobsCorruptedError`. On `main`, `update_cache` inserts the incorrect statepoint into the cache.

Reply:
I found a more localized fix to the test that does not require a public `validate` argument.

Reply:
Very helpful, thanks!